tests/test_runner/test_hooks.py

```diff
@@ -355,6 +356,105 @@ def test_cosine_runner_hook(multi_optimziers):
     hook.writer.add_scalars.assert_has_calls(calls, any_order=True)
 
 
+@pytest.mark.parametrize('multi_optimziers', (True, False))
+def test_flat_cosine_runner_hook(multi_optimziers):
+    """xdoctest -m tests/test_hooks.py test_flat_cosine_runner_hook."""
+    sys.modules['pavi'] = MagicMock()
+    loader = DataLoader(torch.ones((10, 2)))
+    runner = _build_demo_runner(multi_optimziers=multi_optimziers)
+
+    with pytest.raises(AssertionError):
+        # by_epoch should be False
+        FlatCosineAnnealingLrUpdaterHook(by_epoch=True)
```

Comment:
Does the FlatCosineAnnealingLrUpdaterHook only support `by_epoch=True`

Reply:
Both should be supported, but in general `by_epoch=False` is recommended.

Reply:
so why `by_epoch=True` will raise `AssertionError`?

Reply:
I thought it was similar to `test_one_cycle_runner_hook`, so I just copied the code from it.

Reply:
In fact, `OneCycleLrUpdaterHook` only support `by_epoch=False`

https://github.com/open-mmlab/mmcv/blob/bf2c9fa8d21e4af903769b3d7aca20452e4b4669/mmcv/runner/hooks/lr_updater.py#L488-L492

Reply:
OK. I will remove this test.

Reply:
ok